File: packages/sdk-php/src/AiBrainClient.php
<?php
declare(strict_types=1);

namespace AiBrain\Sdk;

/** Thin PHP REST wrapper — no business logic (Phase 16). */
final class AiBrainClient
{
    public function __construct(
        private readonly string $baseUrl,
        private readonly ?string $apiKey = null,
    ) {}

    public function getCapabilities(): array
    {
        return $this->request('GET', '/api/v1/capabilities', auth: false);
    }

    public function searchMemories(string $query): array
    {
        return $this->request('GET', '/api/v1/search?q=' . urlencode($query));
    }

    /** @return array<string, mixed> */
    private function request(string $method, string $path, ?array $body = null, bool $auth = true): array
    {
        $url = rtrim($this->baseUrl, '/') . $path;
        $headers = ['Accept: application/json'];
        if ($auth && $this->apiKey) {
            $headers[] = 'Authorization: Bearer ' . $this->apiKey;
            $headers[] = 'X-API-Key: ' . $this->apiKey;
        }
        $opts = [
            'http' => [
                'method' => $method,
                'header' => implode("\r\n", $headers),
                'ignore_errors' => true,
            ],
        ];
        if ($body !== null) {
            $opts['http']['header'] .= "\r\nContent-Type: application/json";
            $opts['http']['content'] = json_encode($body);
        }
        $ctx = stream_context_create($opts);
        $raw = file_get_contents($url, false, $ctx);
        return json_decode($raw ?: '{}', true) ?? [];
    }
}

File: packages/sdk-java/src/main/java/com/aibrain/sdk/AiBrainClient.java
package com.aibrain.sdk;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

/** Thin Java REST wrapper — no business logic (Phase 16). */
public final class AiBrainClient {
  private final String baseUrl;
  private final String apiKey;
  private final HttpClient http = HttpClient.newHttpClient();

  public AiBrainClient(String baseUrl, String apiKey) {
    this.baseUrl = baseUrl.replaceAll("/$", "");
    this.apiKey = apiKey;
  }

  public String getCapabilities() throws Exception {
    var req = HttpRequest.newBuilder(URI.create(baseUrl + "/api/v1/capabilities")).GET().build();
    return http.send(req, HttpResponse.BodyHandlers.ofString()).body();
  }

  public String searchMemories(String query) throws Exception {
    var req = HttpRequest.newBuilder(URI.create(baseUrl + "/api/v1/search?q=" + query))
        .header("Authorization", "Bearer " + apiKey)
        .header("X-API-Key", apiKey)
        .GET()
        .build();
    return http.send(req, HttpResponse.BodyHandlers.ofString()).body();
  }
}
